tunnel/IpTunnel.c

```diff
@@ -105,6 +105,33 @@ static struct IpTunnel_Connection* newConnection(bool isOutgoing, struct IpTunne
     return conn;
 }
 
+static void deleteConnection(struct IpTunnel_Connection* conn, struct IpTunnel_pvt* context)
+{
+    // Delete connection and shift the list elements following the removed connection
+
+    int i = (((char *)conn)-((char *)&context->pub.connectionList.connections[0]))
+                                                  / sizeof(struct IpTunnel_Connection);
+
+    // Sanity check
+    if (i < 0 || i >= (signed int)context->pub.connectionList.count) {
```

Comment:
Unless you think there's a good reason why it might be false you should  Assert_true()